easybuild/framework/easyblock.py

```diff
@@ -1614,108 +1634,128 @@ def make_module_group_check(self):
 
         return txt
 
-    def make_module_req(self):
+    def make_module_req(self, fake=False):
         """
-        Generate the environment-variables to run the module.
+        Generate the environment-variables required to run the module.
+        Fake modules can set search paths to empty directories.
         """
-        requirements = self.make_module_req_guess()
-
-        lines = ['\n']
-        if os.path.isdir(self.installdir):
-            old_dir = change_dir(self.installdir)
-        else:
-            old_dir = None
+        mod_lines = ['\n']
 
         if self.dry_run:
             self.dry_run_msg("List of paths that would be searched and added to module file:\n")
             note = "note: glob patterns are not expanded and existence checks "
             note += "for paths are skipped for the statements below due to dry run"
-            lines.append(self.module_generator.comment(note))
-
-        # For these environment variables, the corresponding directory must include at least one file.
-        # The values determine if detection is done recursively, i.e. if it accepts directories where files
-        # are only in subdirectories.
-        keys_requiring_files = {
-            'PATH': False,
-            'LD_LIBRARY_PATH': False,
-            'LIBRARY_PATH': True,
-            'CPATH': True,
-            'CMAKE_PREFIX_PATH': True,
-            'CMAKE_LIBRARY_PATH': True,
-        }
+            mod_lines.append(self.module_generator.comment(note))
+
+        if self.make_module_req_guess.__qualname__ != "EasyBlock.make_module_req_guess":
+            # Deprecated make_module_req_guess method used in child Easyblock
+            # Update environment with custom make_module_req_guess
+            self.log.deprecated(
+                "make_module_req_guess() is deprecated, use EasyBlock.module_load_environment instead.",
+                "6.0",
+            )
+            self.module_load_environment.update(self.make_module_req_guess())
+
+        # Only inject path-like environment variables into module file
+        env_var_requirements = sorted([
+            (envar_name, envar_val) for envar_name, envar_val in self.module_load_environment.items()
+            if envar_val.is_path
```

Comment:
@lexming Shouldn't we at least log a warning or something if any entries in `self.module_load_environment` do not have `is_path` is True?

Because those cases will essentially be totally silently ignores currently?

Reply:
Yep we can log that, not sure if this deserves a warning though. I guess there might be cases where no path is added to the module file that are ok, like a `ModuleRC`.

Reply:
My suggestion relates more to entries in `self.module_load_environment` for which `is_path` is `False`, those cases should be exposed clearly rather than silently ignoring them?

Reply:
@lexming Since this is a minor thing that only concerns logging, let's follow up on that in a separate PR, I didn't want to hold back this PR any longer...

Reply:
Opened a new PR for this, see #4748